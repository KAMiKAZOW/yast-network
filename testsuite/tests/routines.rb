# encoding: utf-8

#***************************************************************************
#
# Copyright (c) 2012 Novell, Inc.
# All Rights Reserved.
#
# This program is free software; you can redistribute it and/or
# modify it under the terms of version 2 of the GNU General Public License as
# published by the Free Software Foundation.
#
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
# GNU General Public License for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program; if not, contact Novell, Inc.
#
# To contact Novell about this file by physical or electronic mail,
# you may find current contact information at www.novell.com
#
#**************************************************************************
module Yast
  class RoutinesClient < Client
    def main
      Yast.import "UI"
      Yast.import "Testsuite"
      @READ = { "target" => { "tmpdir" => "/tmp", "stat" => {} } }
      Testsuite.Init([@READ], 0)

      Yast.include self, "testsuite.rb"

      Yast.include self, "network/routines.rb"

      Testsuite.Dump("list2items")
      Testsuite.Test(lambda { list2items(["x", "y"], 0) }, [], nil)

      Testsuite.Dump("hwlist2items")
      Testsuite.Test(lambda do
        hwlist2items([{ "name" => "x" }, { "name" => "y" }], 0)
      end, [], nil)

      nil
    end
  end
end

Yast::RoutinesClient.new.main
