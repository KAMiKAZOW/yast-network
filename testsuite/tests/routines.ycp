{

include "testsuite.ycp";
include "network/routines.ycp";

DUMP("mapkeys");
TEST(``(mapkeys(nil)), [], nil);
TEST(``(mapkeys($[])), [], nil);
TEST(``(mapkeys($[1:2])), [], nil);
TEST(``(mapkeys($["2":3])), [], nil);
TEST(``(mapkeys($[`x:4])), [], nil);
TEST(``(mapkeys($[1:2,3:4])), [], nil);

DUMP("mapvalues");
TEST(``(mapvalues(nil)), [], nil);
TEST(``(mapvalues($[])), [], nil);
TEST(``(mapvalues($[1:2])), [], nil);
TEST(``(mapvalues($["2":3])), [], nil);
TEST(``(mapvalues($[`x:4])), [], nil);
TEST(``(mapvalues($[1:2,3:4])), [], nil);

DUMP("mapvalue");
TEST(``(mapvalue(nil, 1)), [], nil);
TEST(``(mapvalue($[], 1)), [], nil);
TEST(``(mapvalue($[1:2], nil)), [], nil);
TEST(``(mapvalue($[1:2], 1)), [], nil);
TEST(``(mapvalue($[1:2], 2)), [], nil);
TEST(``(mapvalue($["2":3], 1)), [], nil);
TEST(``(mapvalue($["2":3], "2")), [], nil);
TEST(``(mapvalue($[`x:4], `x)), [], nil);
TEST(``(mapvalue($[`x:4], `y)), [], nil);
TEST(``(mapvalue($[`x:4], 1)), [], nil);
TEST(``(mapvalue($[1:2,3:4], 1)), [], nil);
TEST(``(mapvalue($[1:2,3:4], 2)), [], nil);
TEST(``(mapvalue($[1:2,3:4], `x)), [], nil);

DUMP("mapcheckkeys");
TEST(``(mapcheckkeys(nil, [1])), [], nil);
TEST(``(mapcheckkeys($[], [1])), [], nil);
TEST(``(mapcheckkeys($[1:2], [nil])), [], nil);
TEST(``(mapcheckkeys($[1:2], [1])), [], nil);
TEST(``(mapcheckkeys($[1:2], [2])), [], nil);
TEST(``(mapcheckkeys($["2":3], [1])), [], nil);
TEST(``(mapcheckkeys($["2":3], ["2"])), [], nil);
TEST(``(mapcheckkeys($[`x:4], [`x])), [], nil);
TEST(``(mapcheckkeys($[`x:4], [`y])), [], nil);
TEST(``(mapcheckkeys($[`x:4], [1])), [], nil);
TEST(``(mapcheckkeys($[1:2,3:4], [1])), [], nil);
TEST(``(mapcheckkeys($[1:2,3:4], [1,3])), [], nil);
TEST(``(mapcheckkeys($[1:2,3:4], [3,1])), [], nil);
TEST(``(mapcheckkeys($[1:2,3:4], [2,3])), [], nil);
TEST(``(mapcheckkeys($[1:2,3:4], [4,5])), [], nil);
TEST(``(mapcheckkeys($[1:2,3:4], [2,1])), [], nil);
TEST(``(mapcheckkeys($[1:2,3:4], [`x])), [], nil);

DUMP("list2items");
TEST(``(list2items(["x","y"], 0)), [], nil);

DUMP("hwlist2items");
TEST(``(hwlist2items([$["name":"x"],$["name":"y"]], 0)), [], nil);

DUMP("device_type");
TEST(``(device_type("eth0")), [], nil);
TEST(``(device_type("eth1")), [], nil);
TEST(``(device_type("eth-pcmcia-0")), [], nil);
TEST(``(device_type("tr-pcmcia-1")), [], nil);
TEST(``(device_type("ippp2")), [], nil);
TEST(``(device_type("ppp2")), [], nil);

DUMP("device_num");
TEST(``(device_num("eth0")), [], nil);
TEST(``(device_num("eth1")), [], nil);
TEST(``(device_num("eth-pcmcia-0")), [], nil);
TEST(``(device_num("eth-usb-1")), [], nil);
TEST(``(device_num("tr-pcmcia-2")), [], nil);
TEST(``(device_num("ippp2")), [], nil);
TEST(``(device_num("ppp2")), [], nil);

DUMP("device_name");
TEST(``(device_name("eth",0)), [], nil);
TEST(``(device_name("eth",1)), [], nil);
TEST(``(device_name("eth-pcmcia-",0)), [], nil);
TEST(``(device_name("tr-pcmcia-",1)), [], nil);
TEST(``(device_name("ippp",2)), [], nil);
TEST(``(device_name("ppp",2)), [], nil);

}
