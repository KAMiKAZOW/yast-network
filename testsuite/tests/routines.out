Dump	mapkeys
Return	[]
Return	[]
Return	[1]
Return	["2"]
Return	[`x]
Return	[1, 3]
Dump	mapvalues
Return	[]
Return	[]
Return	[2]
Return	[3]
Return	[4]
Return	[2, 4]
Dump	mapvalue
Return	nil
Return	nil
Return	nil
Return	2
Return	nil
Return	nil
Return	3
Return	4
Return	nil
Return	nil
Return	2
Return	nil
Return	nil
Dump	mapcheckkeys
Return	false
Log	Missing key: 1
Return	false
Log	Missing key: nil
Return	false
Return	true
Log	Missing key: 2
Return	false
Log	Missing key: 1
Return	false
Return	true
Return	true
Log	Missing key: `y
Return	false
Log	Missing key: 1
Return	false
Return	true
Return	true
Return	true
Log	Missing key: 2
Return	false
Log	Missing key: 4
Log	Missing key: 5
Return	false
Log	Missing key: 2
Return	false
Log	Missing key: `x
Return	false
Dump	list2items
Return	[`item (`id (0), "x", true), `item (`id (1), "y", false)]
Dump	hwlist2items
Return	[`item (`id (0), "x", true), `item (`id (1), "y", false)]
Dump	device_type
Return	eth
Return	eth
Return	eth-pcmcia-
Return	eth-usb-
Return	tr-pcmcia-
Return	ippp
Return	ppp
Return	lo
Return	eth
Return	eth
Return	eth-pcmcia-
Return	eth-usb-
Return	tr-pcmcia-
Return	ippp
Return	ppp
Return	lo
Dump	device_num
Return	0
Return	1
Return	0
Return	1
Return	2
Return	2
Return	2
Return	0
Return	1
Return	0
Return	1
Return	2
Return	2
Return	2
Return	1
Return	
Dump	alias_num
Return	1
Return	20
Return	3
Return	0
Return	432
Return	2
Return	2
Return	1
Dump	device_name
Return	eth0
Return	eth1
Return	eth-pcmcia-0
Return	tr-pcmcia-1
Return	ippp2
Return	ppp2
Return	lo
Dump	alias_name
Return	eth0:5
Return	eth1:4
Return	eth-pcmcia-0:1
Return	tr-pcmcia-1:1
Return	ippp2:0
Return	ppp2:0
Return	lo:7
Dump	quote
Return	
Return	
Return	a
Return	a'\''b
Return	a'\''b'\''c
Dump	unquote
Return	
Return	
Return	a
Return	a'b
Return	a'b'c
